### PR TITLE
feat(bindings/python): cursor object for PEP 249

### PR DESCRIPTION
**Pull request by everpcpc**

### CODE REVIEW

File: bindings/nodejs/README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ pnpm run build
 const { Client } = require("databend-driver");
 
 const client = new Client(
-  "databend+http://root:root@localhost:8000/?sslmode=disable",
+  "databend://root:root@localhost:8000/?sslmode=disable",
```
Comment (youngsofun) [inline bindings/nodejs/README.md:17]: why need to change DSN format?

Comment (everpcpc) [reply bindings/nodejs/README.md:17]: `http` is not used by driver, this may be misleading to users.

File: driver/src/rest_api.rs [hunk 2]
```diff
@@ -300,15 +300,19 @@ impl Stream for RestAPIRows {
...
-                    let mut new_data = resp.data.into();
-                    self.data.append(&mut new_data);
                     if self.schema.fields().is_empty() {
                         self.schema = Arc::new(resp.schema.try_into()?);
                     }
                     self.next_uri = resp.next_uri;
                     self.next_page = None;
                     self.stats = Some(ServerStats::from(resp.stats));
-                    self.poll_next(cx)
+                    if resp.data.is_empty() {
```
Comment (everpcpc) [inline driver/src/rest_api.rs:309]: cc @youngsofun Is this OK?

Comment (youngsofun) [reply driver/src/rest_api.rs:309]: if  resp.next_uri is not none && resp.data.is_empty()  ?

Comment (everpcpc) [reply driver/src/rest_api.rs:309]: Then we return Poll::Pending and wait for the next poll.

**Review by youngsofun — APPROVED** [review]

**Pull request merged**